src/shortcuts/__tests__/extractThought.ts

```diff
@@ -0,0 +1,101 @@
+import { screen, findAllByText, findAllByPlaceholderText } from '@testing-library/react'
+import { extractThought, newThought } from '../../action-creators'
+import { store } from '../../store'
+import createTestApp, { cleanupTestApp } from '../../test-helpers/createRtlTestApp'
+import { setCursorFirstMatchActionCreator } from '../../test-helpers/setCursorFirstMatch'
+
+/**
+ * Gets a thought by value.
+ */
+const findThoughtByText = async (value: string, container?: HTMLElement) => {
+  const thoughtNodes = await (container ? findAllByText(container!, value, { exact: true }) : screen.findAllByText(value, { exact: true }))
+  return thoughtNodes.find(t => t.hasAttribute('contenteditable'))
+}
+
+/**
+ * Set range selection.
+ */
+const setSelection = (element: HTMLElement, selectionStart: number, selectionEnd: number) => {
+  const range = document.createRange()
+  const sel = window.getSelection()
+
+  range.setStart(element.childNodes[0], selectionStart)
+  range.setEnd(element.childNodes[0], selectionEnd)
+
+  sel?.removeAllRanges()
+  sel?.addRange(range)
+
+  return range.toString()
+}
+
+describe('Extract thought', () => {
```

Comment:
The `describe` block is mainly useful to group tests within a test module. Since there is only one `describe` block here, you can omit it. The filename will show up in the test output and describes the same information.